package entity;
// Generated 2019-3-19 22:57:03 by Hibernate Tools 5.2.11.Final


import javax.ejb.Stateless;
import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;

/**
 * Home object for domain model class Workflowparticipant.
 * @see entity.Workflowparticipant
 * @author Hibernate Tools
 */
@Stateless
public class WorkflowparticipantHome {

    private static final Log log = LogFactory.getLog(WorkflowparticipantHome.class);

    @PersistenceContext private EntityManager entityManager;
    
    public void persist(Workflowparticipant transientInstance) {
        log.debug("persisting Workflowparticipant instance");
        try {
            entityManager.persist(transientInstance);
            log.debug("persist successful");
        }
        catch (RuntimeException re) {
            log.error("persist failed", re);
            throw re;
        }
    }
    
    public void remove(Workflowparticipant persistentInstance) {
        log.debug("removing Workflowparticipant instance");
        try {
            entityManager.remove(persistentInstance);
            log.debug("remove successful");
        }
        catch (RuntimeException re) {
            log.error("remove failed", re);
            throw re;
        }
    }
    
    public Workflowparticipant merge(Workflowparticipant detachedInstance) {
        log.debug("merging Workflowparticipant instance");
        try {
            Workflowparticipant result = entityManager.merge(detachedInstance);
            log.debug("merge successful");
            return result;
        }
        catch (RuntimeException re) {
            log.error("merge failed", re);
            throw re;
        }
    }
    
    public Workflowparticipant findById( Integer id) {
        log.debug("getting Workflowparticipant instance with id: " + id);
        try {
            Workflowparticipant instance = entityManager.find(Workflowparticipant.class, id);
            log.debug("get successful");
            return instance;
        }
        catch (RuntimeException re) {
            log.error("get failed", re);
            throw re;
        }
    }
}

